source/common/upstream/outlier_detection_impl.cc

```diff
@@ -329,11 +329,17 @@ void DetectorImpl::checkHostForUneject(HostSharedPtr host, DetectorHostMonitorIm
     return;
```

Comment:
I believe this is the reason why below the optional is not checked. But I agree that we should assert in that case.